google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -127,40 +113,43 @@ private Publisher(Builder builder) throws IOException {
     cachedTopicNameString = topicName.toString();
 
     this.batchingSettings = builder.batchingSettings;
-    this.retrySettings = builder.retrySettings;
-    this.longRandom = builder.longRandom;
 
     messagesBatch = new LinkedList<>();
     messagesBatchLock = new ReentrantLock();
     activeAlarm = new AtomicBoolean(false);
     executor = builder.executorProvider.getExecutor();
     if (builder.executorProvider.shouldAutoClose()) {
-      closeables.add(new ExecutorAsBackgroundResource(executor));
-    }
-    TransportChannelProvider channelProvider = builder.channelProvider;
-    if (channelProvider.needsExecutor()) {
-      channelProvider = channelProvider.withExecutor(executor);
+      closeables =
+          Collections.<AutoCloseable>singletonList(new ExecutorAsBackgroundResource(executor));
+    } else {
+      closeables = Collections.emptyList();
```

Comment:
Why not just pass `executorProvider` to the client and let the client close it down? (It already has to close down the channel)

Reply:
`Publisher` itself needs to use the executor to schedule publishing. AFAIK, the stub can't expose the executor it's currently using right?

Reply:
Ahh good point.